lib/__tests__/standalone-cache.test.js

```diff
@@ -0,0 +1,146 @@
+
+  beforeEach(() => {
+    // Initial run to warm up the cache
+    return standalone(getConfig())
+  })
+
+  afterEach(() => {
+    // Clean up after each test case
+    fsExtra.removeSync(expectedCacheFilePath)
+    fsExtra.removeSync(changedFile)
+  })
+
+  it("cache file is created at $CWD/.stylelintcache", () => {
+    // Ensure cache file exists
+    expect(fileExists(expectedCacheFilePath)).toBe(true)
+    const cacheFile = fsExtra.readJsonSync(expectedCacheFilePath)
+    // Ensure cache file contains only linted css file
+    expect(typeof cacheFile[validFile] === "object").toBe(true)
+    expect(typeof cacheFile[changedFile] === "undefined").toBe(true)
+  })
+
+  it("only changed files are linted", () => {
+    // Add "changed" file
+    fsExtra.copySync(validFile, changedFile)
+    // Next run should lint only changed files
+    return standalone(getConfig()).then(output => {
+      // Ensure only changed files are linted
+      const isValidFileLinted = !!output.results.find(file => file.source === validFile)
+      const isChangedFileLinted = !!output.results.find(file => file.source === changedFile)
+      expect(isValidFileLinted).toBe(false)
+      expect(isChangedFileLinted).toBe(true)
+      // Ensure cache file contains linted css files
+      const cachedFiles = fsExtra.readJsonSync(expectedCacheFilePath)
+      expect(typeof cachedFiles[validFile] === "object").toBe(true)
+      expect(typeof cachedFiles[changedFile] === "object").toBe(true)
+    })
+  })
+
+  it("all files are linted on config change", () => {
+    fsExtra.copySync(validFile, changedFile)
+    const changedConfig = getConfig()
+    changedConfig.config.rules["block-no-empty"] = false
+    // All file should be re-linted as config has changed
+    return standalone(changedConfig).then(output => {
+      // Ensure all files are re-linted
+      const isValidFileLinted = !!output.results.find(file => file.source === validFile)
+      const isChangedFileLinted = !!output.results.find(file => file.source === changedFile)
+      expect(isValidFileLinted).toBe(true)
+      expect(isChangedFileLinted).toBe(true)
+    })
+  })
+
+  it("invalid files are not cached", () => {
+    fsExtra.copySync(invalidFile, changedFile)
+    // Should lint only changed files
+    return standalone(getConfig()).then((output) => {
+      expect(output.errored).toBe(true)
+      // Ensure only changed files are linted
+      const isValidFileLinted = !!output.results.find(file => file.source === validFile)
+      const isInvalidFileLinted = !!output.results.find(file => file.source === changedFile)
+      expect(isValidFileLinted).toBe(false)
+      expect(isInvalidFileLinted).toBe(true)
+      // Ensure cache file doesn't contain invalid file
+      const cachedFiles = fsExtra.readJsonSync(expectedCacheFilePath)
+      expect(typeof cachedFiles[validFile] === "object").toBe(true)
+      expect(typeof cachedFiles[changedFile] === "undefined").toBe(true)
```

Comment:
I'll update the var as per [comment](https://github.com/stylelint/stylelint/pull/2293#discussion_r102703828)